manifests/components/kibana.jsonnet

```diff
@@ -38,18 +39,66 @@ local strip_trailing_slash(s) = (
     },
   },
 
+  // List of Kibana plug-ins to install
+  plugins:: {
+    /* Example:
+    logtrail: {
+      version: "0.1.31",
+      url: "https://github.com/sivasamyk/logtrail/releases/download/v0.1.31/logtrail-6.7.1-0.1.31.zip",
+    },
+    */
+  },
+
   es: error "elasticsearch is required",
 
   serviceAccount: kube.ServiceAccount($.p + "kibana") + $.metadata {
   },
 
+  pvc: kube.PersistentVolumeClaim($.p + "kibana-plugins") + $.metadata {
+    storage: "2Gi",
+  },
+
   deploy: kube.Deployment($.p + "kibana") + $.metadata {
     spec+: {
       template+: {
         spec+: {
           securityContext: {
             fsGroup: 1001,
           },
+          volumes_+: {
+            plugins: kube.PersistentVolumeClaimVolume($.pvc),
+          },
+          initContainers_+: {
+            kibana_plugins_install: kube.Container("kibana-plugins-install") {
+              image: KIBANA_IMAGE,
+              securityContext: {
+                allowPrivilegeEscalation: false,
+              },
+              local wanted = std.join("\n", ["%s@%s,%s" % [k, $.plugins[k].version, $.plugins[k].url] for k in std.objectFields($.plugins)]),
+              command: [
+                "/bin/sh",
+                "-c",
+                |||
+                  set -e
+                  rm -rf /opt/bitnami/kibana/plugins/lost+found
+                  echo %s | sort > /tmp/wanted.list
+                  /opt/bitnami/kibana/bin/kibana-plugin list | grep @ | sort > /tmp/installed.list
+                  join -v2 -t, -j1 /tmp/wanted.list /tmp/installed.list | while read plugin; do
+                    ${plugin:+/opt/bitnami/kibana/bin/kibana-plugin remove "${plugin%%@*}"}
```

Comment:
updated to account for empty list /cc @anguslees

Reply:
... aha!  I was confused because it _should_ have done-the-right-thing with empty lists.  The issue is/was that the "echo %s | sort >/tmp/wanted.list" would create a list with an empty line, because `echo ""`.

So another solution (with hindsight) would have been to do `echo %s | grep . | sort >/tmp/wanted.list` or a more sophisticated `echo` variation.  (Not a suggestion to change, I just wanted to understand the underlying issue)